test/Serilog.PerformanceTests/AlmostRealWorldBenchmark.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+using Serilog.Context;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.PerformanceTests.Support;
+
+namespace Serilog.PerformanceTests
+{
+    [MemoryDiagnoser]
+    [ShortRunJob]
+    public class AlmostRealWorldBenchmark
+    {
+        static readonly LoggingLevelSwitch LoggingLevelSwitch = new LoggingLevelSwitch(LogEventLevel.Verbose);
+        static readonly Random Rnd = new Random(42);
+
+        ILogger CreateLog()
+        {
+            return Log.Logger = new LoggerConfiguration()
+                .Enrich.FromLogContext()
+                .Enrich.WithProperty("Environment", "BenchmarkTester")
+                .Enrich.WithProperty("Application", "Serilog.PerformanceTests.AlmostRealWorldBenchmark")
+                .Enrich.WithProperty("ExecutionId", Guid.NewGuid())
+                .Enrich.WithProperty("ExecutionStartDate", DateTimeOffset.UtcNow)
+                .MinimumLevel.ControlledBy(LoggingLevelSwitch)
+                .WriteTo.Logger(l => l
+                    .Filter.ByExcluding(ev => ev.Properties.ContainsKey("ProgressEntry"))
+                    .WriteTo.Sink(new NullSink())) //To Simulate a FileSink that don't 'records' some LogEvents
+                .WriteTo.Sink(new NullSink()) //To Simulate a Console Sink that show all LogEvents
+                .CreateLogger();
+        }
+
+        [Benchmark(Baseline = true)]
+        public void LogLikeAApp()
+        {
+            var log = CreateLog(); //Always create a new instance of the logger each test to Benchmark the creation with the test.
```

Comment:
This might be better separated into its own benchmark, since we'd normally strongly recommend against creating more than one or two logging pipelines during an application run.

Reply:
Sorry, @nblumhardt. I didn't understand what you meant with "separated into its own benchmark". 
This is already a new benchmark. Can you explain better to me? Thanks.

Reply:
I know about the recommendation, but what I want to test/benchmark in this test is exactly the performance and the "cost" of using Serilog in a "real-world" scenario. Including the creation of the pipelines.

My idea is this entire benchmark simulates a console application that uses Serilog. 
In my previous tests this was a Console application thar I was using to profile. https://github.com/rafaelsc/serilog/tree/refactorLogger/TestConsole

Reply:
Now I think what is missing in this AlmostRealWorldBenchmark is another scenario without Serilog, but with the same code. That way we can measure the "cost" of implementation Serilog in an application. I will create this new test.

Reply:
Hi @rafaelsc, thanks for your reply; I think the issue with the `AlmostRealWorld` scenario is that in real-world apps, the pipeline is configured once, while the logging methods are called may times.

If the benchmark were split into a "pipeline creation" benchmark then this would show the cost of setting up Serilog, separately from the cost of writing log events. As it is now, the mix of methods does benchmark these things, but not in a way that is representative of how Serilog usually operates in real-world apps.